### ISSUE TITLE
False positive of `IfUnlessModifier` for array elements

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop-hq/rubocop#8226]
Hello. I've found a false positive of `IfUnlessModifier` for array elements with conditions. 

You can write an array element with multi-line condition, and with one-line condition, but only with parentheses, without them there is a syntax error:

```
unexpected `if' modifier, expecting ']'
```

But `IfUnlessModifier` doesn't count this specific condition (for conditions), and offenses a multi-line condition, which can be placed into one line with syntax error, but with parentheses it already exceeds `LineLength`.

--------

## Expected behavior

No offenses for multi-line condition in array, which can't be placed into one line without `LineLength` offense.

## Actual behavior

There is a "fork" with offenses in any case.

## Steps to reproduce the problem

```yaml
Layout/LineLength:
  Max: 12
Style/FrozenStringLiteralComment:
  Enabled: false
```

```ruby
[
  if foo
    bar
  end
]
```

```
rubocop/test.rb:2:3: C: Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
  if foo
  ^^
```

```ruby
[
  (bar if foo)
]
```

```
rubocop/test.rb:2:8: C: Style/IfUnlessModifier: Modifier form of if makes the line too long.
  (bar if foo)
       ^^
rubocop/test.rb:2:13: C: Layout/LineLength: Line is too long. [14/12]
  (bar if foo)
            ^^
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
0.86.0 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.7.1 x86_64-linux)
```
